modules/lib/src/main/resources/assets/js/v6/features/shared/dialogs/new-content/NewContentDialog.tsx

```diff
@@ -0,0 +1,163 @@
+import {Dialog, Tab} from '@enonic/ui';
+import {ReactElement, useRef, useState} from 'react';
+import {
+    $newContentDialog,
+    closeNewContentDialog,
+    setInputValue,
+    setSelectedTab,
+} from '../../../store/dialogs/newContentDialog.store';
+import {useStore} from '@nanostores/preact';
+import {KeyHelper} from '@enonic/lib-admin-ui/ui/KeyHelper';
+import {NewContentDialogMediaTab} from './NewContentDialogMediaTab';
+import {NewContentDialogContentTypesTab} from './NewContentDialogContentTypesTab';
+import {uploadDragImages, uploadMediaFiles} from './NewContentDialogUpload';
+import {NewContentDialogSearch} from './NewContentDialogSearch';
+import {useI18n} from '../../../hooks/useI18n';
+
+const NEW_CONTENT_DIALOG_NAME = 'NewContentDialog';
+
+export const NewContentDialog = (): ReactElement => {
+    const [isInputVisible, setIsInputVisible] = useState(false);
+    const inputRef = useRef<HTMLInputElement>(null);
+    const dialogContentRef = useRef<HTMLDivElement>(null);
+    const {open, selectedTab, parentContent, filteredBaseContentTypes, filteredSuggestedContentTypes} =
+        useStore($newContentDialog);
+
+    const titleLabel = useI18n('dialog.new.title');
+    const allTabLabel = useI18n('dialog.new.tab.all');
+    const suggestedTabLabel = useI18n('dialog.new.tab.suggested');
+    const mediaTabLabel = useI18n('dialog.new.tab.media');
+    const hintLabel = useI18n('dialog.new.hint.filter');
+
+    //
+    // * Labels
+    //
```

Comment:
Again, more of a stylistic recommendation: use such comments only on root level, not inside the component. Too much visual noise for insignificant differences of code blocks.